services/private_ip_manager/src/main/java/com/futurewei/alcor/privateipmanager/repo/IpAddrRangeRepo.java

```diff
@@ -190,17 +196,9 @@ private IpAddrAlloc doAllocateIpAddr(String vpcId, int ipVersion, String ipAddr)
      * @throws Exception Db operation or ip address assignment exception
      */
     @DurationStatistics
-    public synchronized IpAddrAlloc allocateIpAddr(IpAddrRequest request) throws Exception {
-        IpAddrAlloc ipAddrAlloc = null;
-        try (Transaction tx = ipAddrRangeCache.getTransaction().start()) {
-            ipAddrAlloc = allocateIpAddrMethod(request);
-            tx.commit();
+    public IpAddrAlloc allocateIpAddr(IpAddrRequest request) throws Exception {
+            IpAddrAlloc ipAddrAlloc = allocateIpAddrMethod(request);
```

Comment:
@DavidLiu506 Do we still need this function if we push all the process code into `allocateIpAddrMethod`?

Reply:
It called by other functions.